File: ruby/matrix/matrix.rb
=begin
Write your code for the 'Matrix' exercise in this file. Make the tests in
`matrix_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/matrix` directory.
=end

class Matrix
  attr_reader :string

  def initialize(string)
    @string = string
  end

  def rows
    string.split("\n").map { |row| row.split.map(&:to_i) }
  end

  def columns
    rows.transpose
  end
end

File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(string) {
    this.string = string;
    // throw new Error("Remove this statement and implement this function");
  }

  get rows() {
    return this.string.split(/\r?\n/).map(row => row.split(" ").map(e => Number(e)));
    // throw new Error("Remove this statement and implement this function");
  }

  get columns() {
    let rows = this.string.split(/\r?\n/).map(row => row.split(" ").map(e => Number(e)));
    return rows[0].map((col, i) => rows.map(row => row[i]));
  }
}
